plugins/module_utils/foreman_helper.py

```diff
@@ -622,7 +623,8 @@ def wait_for_task(self, task):
 
             resource_payload = self._resource_prepare_params('foreman_tasks', 'show', {'id': task['id']})
             task = self._resource_call('foreman_tasks', 'show', resource_payload)
-
+        if not ignore_errors and task['result'] != 'success':
+            self.fail_json(msg='Task {0} did not succeed. Task information: {1}'.format(task['action'], task['humanized']['errors']))
```

Comment:
When the task times out, we use the task ID in the error message, but I also see a value of printing the action. Would you mind updating both places to read something like "Task ACTION (ID)"?